File: SalaryCounter.Api/ClientApp/src/app/models/worker.ts
import Util from "../util/util";
import {WorkerType} from "./worker.type";

export class Worker {
    id: string;
    name: string;
    employmentDate: Date;
    salaryBase: number;
    workerType: WorkerType;
    chief: string;

    constructor() {
        this.id = Util.getEmptyId();
        this.name = "";
        this.salaryBase = 1000;
        this.employmentDate = new Date();
    }

    static fromData(other: any): Worker {
        let worker = new Worker();
        worker.id = other.id || worker.id;
        worker.name = other.name || worker.name;
        worker.salaryBase = other.salaryBase && Number(other.salaryBase) || worker.salaryBase;
        worker.employmentDate = other.employmentDate && new Date(other.employmentDate) || worker.employmentDate;
        worker.workerType = other.workerType && WorkerType.fromData(other.workerType) || undefined;
        worker.chief = other.chief || worker.chief;
        return worker;
    }

    static fromDataList(other: any[]): Worker[] {
        return other.map(function (worker: any, index: Number, workers: any[]) {
            return Worker.fromData(worker);
        })
    }
}

File: SalaryCounter.Service/Model/Worker.cs
using System;

namespace SalaryCounter.Service.Model
{
    public sealed class Worker
    {
        public Worker(string name, DateTime employmentDate, decimal salaryBase, WorkerType workerType, Guid? id = null, Guid? chief = null)
        {
            Name = name;
            EmploymentDate = employmentDate;
            SalaryBase = salaryBase;
            WorkerType = workerType;
            Id = id ?? Guid.Empty;
            Chief = chief;
        }

        public Guid Id { get; set; }
        public string Name { get; set; }
        public DateTime EmploymentDate { get; set; }
        public decimal SalaryBase { get; set; }
        public WorkerType WorkerType { get; set; }
        public Guid? Chief { get; set; }
    }
}